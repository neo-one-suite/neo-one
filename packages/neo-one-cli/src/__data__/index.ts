export * from './getClients';
export * from './getContracts';
export * from './buildTests';
export * from './generateTests';
