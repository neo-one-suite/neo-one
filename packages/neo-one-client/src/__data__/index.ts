export * from './keys';
export * from './transactions';
