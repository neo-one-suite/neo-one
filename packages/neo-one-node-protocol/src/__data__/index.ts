export { context } from './context';
export { keys } from './keys';
