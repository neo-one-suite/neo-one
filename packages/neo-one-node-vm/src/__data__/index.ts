import * as assets from './assets';
import * as transactions from './transactions';
import * as testUtils from './utils';

export * from './createBlockchain';
export * from './keys';

export { assets, testUtils, transactions };
