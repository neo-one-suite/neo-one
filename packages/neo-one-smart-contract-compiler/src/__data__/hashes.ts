export const hashes = {
  OLD_NEO_ASSET_HASH: '0xc56f33fc6ecfcd0c225c4ab356fee59390af8560be0e930faebe74a6daff7c9b',
  OLD_GAS_ASSET_HASH: '0x602c79718b16e442de58778e148d0b1084e3b2dffd5de6b7b16cee7969282de7',
};
