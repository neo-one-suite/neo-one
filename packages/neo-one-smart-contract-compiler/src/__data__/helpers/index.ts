export * from './execute';
export * from './extractors';
export * from './startNode';
