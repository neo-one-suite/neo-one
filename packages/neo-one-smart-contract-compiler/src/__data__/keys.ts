import { common } from '@neo-one/client-core';

export const keys = [
  {
    name: 'first',
    address: 'ALq7AWrhAueN6mJNqk6FHJjnsEoPRytLdW',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('7d128a6d096f0c14c3a25a2b0c41cf79661bfcb4a8cc95aaaea28bde4d732344', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('02028a99826edc0c97d18e22b6932373d908d323aa7f92656a77ec26e8861699ef', 'hex'),
    ),
    wif: 'L1QqQJnpBwbsPGAuutuzPTac8piqvbR1HRjrY5qHup48TBCBFe4g',
    password: 'city of zion',
    encryptedWIF: '6PYLHmDf6AjF4AsVtosmxHuPYeuyJL3SLuw7J1U8i7HxKAnYNsp61HYRfF',
    scriptHash: common.bufferToUInt160(Buffer.from('3775292229eccdf904f16fff8e83e7cffdc0f0ce', 'hex')),
  },
  {
    name: 'second',
    address: 'ALfnhLg7rUyL6Jr98bzzoxz5J7m64fbR4s',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('9ab7e154840daca3a2efadaf0df93cd3a5b51768c632f5433f86909d9b994a69', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c9', 'hex'),
    ),
    wif: 'L2QTooFoDFyRFTxmtiVHt5CfsXfVnexdbENGDkkrrgTTryiLsPMG',
    password: '我的密码',
    encryptedWIF: '6PYWVp3xfgvnuNKP7ZavSViYvvim2zuzx9Q33vuWZr8aURiKeJ6Zm7BfPQ',
    scriptHash: common.bufferToUInt160(Buffer.from('35b20010db73bf86371075ddfba4e6596f1ff35d', 'hex')),
  },
  {
    name: 'third',
    address: 'AVf4UGKevVrMR1j3UkPsuoYKSC4ocoAkKx',
    privateKey: common.bufferToPrivateKey(
      Buffer.from('3edee7036b8fd9cef91de47386b191dd76db2888a553e7736bb02808932a915b', 'hex'),
    ),
    publicKey: common.bufferToECPoint(
      Buffer.from('02232ce8d2e2063dce0451131851d47421bfc4fc1da4db116fca5302c0756462fa', 'hex'),
    ),
    wif: 'KyKvWLZsNwBJx5j9nurHYRwhYfdQUu9tTEDsLCUHDbYBL8cHxMiG',
    password: 'MyL33tP@33w0rd',
    encryptedWIF: '6PYNoc1EG5J38MTqGN9Anphfdd6UwbS4cpFCzHhrkSKBBbV1qkbJJZQnkn',
    scriptHash: common.bufferToUInt160(Buffer.from('9847e26135152874355e324afd5cc99f002acb33', 'hex')),
  },
];
