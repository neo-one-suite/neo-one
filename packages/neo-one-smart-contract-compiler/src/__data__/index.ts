import * as helpers from './helpers';

export { helpers };
export * from './keys';
