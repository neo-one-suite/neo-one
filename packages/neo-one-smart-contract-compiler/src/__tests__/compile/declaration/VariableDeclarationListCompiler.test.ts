import { helpers } from '../../../__data__';

describe('VariableDeclarationListCompiler', () => {
  test('binds let variables', async () => {
    await helpers.executeString(`
      let x = 0;
      if (x !== 0) {
        throw 'Failure';
      }

      x = x + 1;
      if (x !== 1) {
        throw 'Failure';
      }
    `);
  });

  test('Illegal Assignment: once defined with const, X should not be re-assignable.', async () => {
    await helpers.compileString(
      `
      const x = 0;
      x = 3;
    `,
      { type: 'error' },
    );
  });

  test('binds const variables', async () => {
    await helpers.executeString(`
      const x = 3;

      if (x !== 3) {
        throw 'Failure';
      }

      const y = x + 1;
      if (y !== 4) {
        throw 'Failure';
      }
    `);
  });

  test('binds variables value to output of function', async () => {
    await helpers.executeString(`
      let i = 3;
      function fn(){return ++i};
      const y = fn();

      if (y !== 4) {
        throw 'Failure';
      }
    `);
  });

  test.skip('binds function to variable & function output to variable', async () => {
    await helpers.executeString(`
      let i = 3;
      const fn = function() {
        return ++i;
      };

      if (fn() !== 4) {
        throw 'Failure';
      }
    `);
  });

  test('binds function return value to variable', async () => {
    await helpers.executeString(`
      let i = 2;
      function fn(){
        return ++i;
      }
      const y = fn();
      if (y !== 3) {
        throw 'Failure';
      }
    `);
  });

  test('binds variable of non US charset', async () => {
    await helpers.executeString(`
      let Früh = 2;
      if (Früh !== 2) {
        throw 'Failure';
      }
    `);
  });

  test('Assignment with incongruent typing', async () => {
    await helpers.compileString(
      `
      // error TS2322: Type '"3"' is not assignable to type 'number'.
       const x: number = '3';

    `,
      { type: 'error' },
    );
  });

  test.skip('binds array composed of spread array to variable', async () => {
    await helpers.executeString(`
      var i = [1,2,3,4,5];
      let j = [...i, 10, 20, 30];
      if ((j[3] + i[6]) !== 24) {
        throw 'Failure';
      }
      `);
  });

  test.skip('binds object composed of spread object to variable', async () => {
    await helpers.executeString(`
      var i = {attr: 'value'};
      var o = {...i, attr2: 'value2'};
      if (o.attr !== "value" || o.attr2 != "value2") {
        throw 'Failure';
      }
      `);
  });

  test.skip('binds variable with string template', async () => {
    await helpers.executeString(
      " \
      let i = [1,2,3,4,5]; \
      let j = `${i[1]} times`; \
      if (j !== '2 times') { \
        throw 'Failure'; \
      } \
      ",
    );
  });

  test('binds variable result of addition', async () => {
    await helpers.executeString(`
      let i = [1,2,3,4,5];
      let j = i[0] + i[4] + 4;
      if (j !== 10) {
        throw 'Failure';
      }
      `);
  });

  test.skip('binds variable with concat of string and number', async () => {
    await helpers.executeString(`
      let i = 'thing ' + 123;
      if (i !== "thing 123") {
        throw 'Failure';
      }
      `);
  });

  test.skip('declare n without assignment using const, should not be empty string', async () => {
    await helpers.executeString(`
      const n; // should be syntax error.
      if (n === '') {
        throw 'Failure';
      }
      `);
  });

  test.skip('declare n without assignment using let and var, should be undefined', async () => {
    await helpers.executeString(`
      let a;
      var b;
      if (a !== undefined || b !== undefined) {
          throw 'Failure';
      }
      `);
  });

  test('bind variable with null', async () => {
    await helpers.executeString(`
      let n = null;
      if (n !== null) {
        throw 'Failure';
      }
      `);
  });

  test('bind variable with return value of an anonymous function', async () => {
    await helpers.executeString(`
      let n = (()=>123)();
      if (n !== 123) {
        throw 'Failure';
      }
      `);
  });

  test('bind variable with undefined', async () => {
    await helpers.executeString(`
      let n = undefined;
      if (n !== undefined) {
        throw 'Failure';
      }
      `);
  });

  test('binds variables with pre and post increments and decrements', async () => {
    await helpers.executeString(`
      let h = 5;
      let i = 1;  // 1
      let j = i++; // 1
      let k = ++i; // 3
      let l = h--; // 5
      let o = --h; // 3
      let p = h + i + j + k + l  + o;
      if (p !== 18) {
        throw 'Failure';
      }
      `);
  });

  test('binds addition result of object member, array entry and integer to variable', async () => {
    await helpers.executeString(`
    let i = [1,2,3,4,5];
    let o = {bar:1,baz:2,fiz:3,fuz:4,far:5};
    let j = o.bar + i[3] + 4;
      if (j !== 9) {
        throw 'Failure';
      }
      `);
  });

  test('binds array variable', async () => {
    await helpers.executeString(`
      let i = [1,2,3,4,5];

      if ((i[0] + i[2]) !== 4) {
        throw 'Failure';
      }

      `);
  });
});
