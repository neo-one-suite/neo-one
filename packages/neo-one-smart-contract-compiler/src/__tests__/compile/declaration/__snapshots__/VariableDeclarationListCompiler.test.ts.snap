// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariableDeclarationListCompiler Assignment with incongruent typing 1`] = `"code.ts (3,14): Type '\\"3\\"' is not assignable to type 'number'."`;

exports[`VariableDeclarationListCompiler Illegal Assignment: once defined with const, X should not be re-assignable. 1`] = `"code.ts (3,7): Cannot assign to 'x' because it is a constant or a read-only property."`;
